From bfeea7ddbeae253ae0461f73da6dfd8903f4c5fa Mon Sep 17 00:00:00 2001
From: Johannes Donath <johannesd@torchmind.com>
Date: Wed, 21 Dec 2016 20:50:10 +0100
Subject: [PATCH] Test Commit


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index aa45aa0..60a4332 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -629,6 +629,7 @@ public abstract class MinecraftServer
   }
 
   static protected void main(String[] var0) {
+    // Test
     Bootstrap.register();
 
     try {
-- 
2.11.0.windows.1

