From bb3a3a7d4389697f5c913c294df11aa59cc945d2 Mon Sep 17 00:00:00 2001
From: Johannes Donath <johannesd@torchmind.com>
Date: Wed, 1 Jun 2016 19:52:35 +0200
Subject: [PATCH] Test commit ...


diff --git a/net/minecraft/block/Block.java b/net/minecraft/block/Block.java
index 08a8c25..59fb499 100644
--- a/net/minecraft/block/Block.java
+++ b/net/minecraft/block/Block.java
@@ -1,153 +1,6 @@
 package net.minecraft.block;
 
 import com.google.common.collect.Sets;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Random;
-import javax.annotation.Nullable;
-import net.minecraft.block.BlockAir;
-import net.minecraft.block.BlockAnvil;
-import net.minecraft.block.BlockBanner;
-import net.minecraft.block.BlockBarrier;
-import net.minecraft.block.BlockBeacon;
-import net.minecraft.block.BlockBed;
-import net.minecraft.block.BlockBeetroot;
-import net.minecraft.block.BlockBookshelf;
-import net.minecraft.block.BlockBrewingStand;
-import net.minecraft.block.BlockButtonStone;
-import net.minecraft.block.BlockButtonWood;
-import net.minecraft.block.BlockCactus;
-import net.minecraft.block.BlockCake;
-import net.minecraft.block.BlockCarpet;
-import net.minecraft.block.BlockCarrot;
-import net.minecraft.block.BlockCauldron;
-import net.minecraft.block.BlockChest;
-import net.minecraft.block.BlockChorusFlower;
-import net.minecraft.block.BlockChorusPlant;
-import net.minecraft.block.BlockClay;
-import net.minecraft.block.BlockCocoa;
-import net.minecraft.block.BlockColored;
-import net.minecraft.block.BlockCommandBlock;
-import net.minecraft.block.BlockCompressedPowered;
-import net.minecraft.block.BlockCrops;
-import net.minecraft.block.BlockDaylightDetector;
-import net.minecraft.block.BlockDeadBush;
-import net.minecraft.block.BlockDirt;
-import net.minecraft.block.BlockDispenser;
-import net.minecraft.block.BlockDoor;
-import net.minecraft.block.BlockDoublePlant;
-import net.minecraft.block.BlockDoubleStoneSlab;
-import net.minecraft.block.BlockDoubleStoneSlabNew;
-import net.minecraft.block.BlockDoubleWoodSlab;
-import net.minecraft.block.BlockDragonEgg;
-import net.minecraft.block.BlockDropper;
-import net.minecraft.block.BlockDynamicLiquid;
-import net.minecraft.block.BlockEmptyDrops;
-import net.minecraft.block.BlockEnchantmentTable;
-import net.minecraft.block.BlockEndGateway;
-import net.minecraft.block.BlockEndPortal;
-import net.minecraft.block.BlockEndPortalFrame;
-import net.minecraft.block.BlockEndRod;
-import net.minecraft.block.BlockEnderChest;
-import net.minecraft.block.BlockFarmland;
-import net.minecraft.block.BlockFence;
-import net.minecraft.block.BlockFenceGate;
-import net.minecraft.block.BlockFire;
-import net.minecraft.block.BlockFlowerPot;
-import net.minecraft.block.BlockFrostedIce;
-import net.minecraft.block.BlockFurnace;
-import net.minecraft.block.BlockGlass;
-import net.minecraft.block.BlockGlowstone;
-import net.minecraft.block.BlockGrass;
-import net.minecraft.block.BlockGrassPath;
-import net.minecraft.block.BlockGravel;
-import net.minecraft.block.BlockHalfStoneSlab;
-import net.minecraft.block.BlockHalfStoneSlabNew;
-import net.minecraft.block.BlockHalfWoodSlab;
-import net.minecraft.block.BlockHardenedClay;
-import net.minecraft.block.BlockHay;
-import net.minecraft.block.BlockHopper;
-import net.minecraft.block.BlockHugeMushroom;
-import net.minecraft.block.BlockIce;
-import net.minecraft.block.BlockJukebox;
-import net.minecraft.block.BlockLadder;
-import net.minecraft.block.BlockLever;
-import net.minecraft.block.BlockLilyPad;
-import net.minecraft.block.BlockMelon;
-import net.minecraft.block.BlockMobSpawner;
-import net.minecraft.block.BlockMushroom;
-import net.minecraft.block.BlockMycelium;
-import net.minecraft.block.BlockNetherBrick;
-import net.minecraft.block.BlockNetherWart;
-import net.minecraft.block.BlockNetherrack;
-import net.minecraft.block.BlockNewLeaf;
-import net.minecraft.block.BlockNewLog;
-import net.minecraft.block.BlockNote;
-import net.minecraft.block.BlockObsidian;
-import net.minecraft.block.BlockOldLeaf;
-import net.minecraft.block.BlockOldLog;
-import net.minecraft.block.BlockOre;
-import net.minecraft.block.BlockPackedIce;
-import net.minecraft.block.BlockPane;
-import net.minecraft.block.BlockPistonBase;
-import net.minecraft.block.BlockPistonExtension;
-import net.minecraft.block.BlockPistonMoving;
-import net.minecraft.block.BlockPlanks;
-import net.minecraft.block.BlockPortal;
-import net.minecraft.block.BlockPotato;
-import net.minecraft.block.BlockPressurePlate;
-import net.minecraft.block.BlockPressurePlateWeighted;
-import net.minecraft.block.BlockPrismarine;
-import net.minecraft.block.BlockPumpkin;
-import net.minecraft.block.BlockPurpurSlab;
-import net.minecraft.block.BlockQuartz;
-import net.minecraft.block.BlockRail;
-import net.minecraft.block.BlockRailDetector;
-import net.minecraft.block.BlockRailPowered;
-import net.minecraft.block.BlockRedFlower;
-import net.minecraft.block.BlockRedSandstone;
-import net.minecraft.block.BlockRedstoneComparator;
-import net.minecraft.block.BlockRedstoneLight;
-import net.minecraft.block.BlockRedstoneOre;
-import net.minecraft.block.BlockRedstoneRepeater;
-import net.minecraft.block.BlockRedstoneTorch;
-import net.minecraft.block.BlockRedstoneWire;
-import net.minecraft.block.BlockReed;
-import net.minecraft.block.BlockRotatedPillar;
-import net.minecraft.block.BlockSand;
-import net.minecraft.block.BlockSandStone;
-import net.minecraft.block.BlockSapling;
-import net.minecraft.block.BlockSeaLantern;
-import net.minecraft.block.BlockSilverfish;
-import net.minecraft.block.BlockSkull;
-import net.minecraft.block.BlockSlab;
-import net.minecraft.block.BlockSlime;
-import net.minecraft.block.BlockSnow;
-import net.minecraft.block.BlockSnowBlock;
-import net.minecraft.block.BlockSoulSand;
-import net.minecraft.block.BlockSponge;
-import net.minecraft.block.BlockStainedGlass;
-import net.minecraft.block.BlockStainedGlassPane;
-import net.minecraft.block.BlockStairs;
-import net.minecraft.block.BlockStandingSign;
-import net.minecraft.block.BlockStaticLiquid;
-import net.minecraft.block.BlockStem;
-import net.minecraft.block.BlockStone;
-import net.minecraft.block.BlockStoneBrick;
-import net.minecraft.block.BlockStructure;
-import net.minecraft.block.BlockTNT;
-import net.minecraft.block.BlockTallGrass;
-import net.minecraft.block.BlockTorch;
-import net.minecraft.block.BlockTrapDoor;
-import net.minecraft.block.BlockTripWire;
-import net.minecraft.block.BlockTripWireHook;
-import net.minecraft.block.BlockVine;
-import net.minecraft.block.BlockWall;
-import net.minecraft.block.BlockWallSign;
-import net.minecraft.block.BlockWeb;
-import net.minecraft.block.BlockWorkbench;
-import net.minecraft.block.BlockYellowFlower;
-import net.minecraft.block.SoundType;
 import net.minecraft.block.material.EnumPushReaction;
 import net.minecraft.block.material.MapColor;
 import net.minecraft.block.material.Material;
@@ -167,13 +20,7 @@ import net.minecraft.item.ItemBlock;
 import net.minecraft.item.ItemStack;
 import net.minecraft.stats.StatList;
 import net.minecraft.tileentity.TileEntity;
-import net.minecraft.util.EnumBlockRenderType;
-import net.minecraft.util.EnumFacing;
-import net.minecraft.util.EnumHand;
-import net.minecraft.util.Mirror;
-import net.minecraft.util.ObjectIntIdentityMap;
-import net.minecraft.util.ResourceLocation;
-import net.minecraft.util.Rotation;
+import net.minecraft.util.*;
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.RayTraceResult;
@@ -184,6 +31,11 @@ import net.minecraft.world.Explosion;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
 
+import javax.annotation.Nullable;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+
 public class Block {
    private static final ResourceLocation AIR_ID = new ResourceLocation("air");
    public static final RegistryNamespacedDefaultedByKey<ResourceLocation, Block> REGISTRY = new RegistryNamespacedDefaultedByKey(AIR_ID);
@@ -215,6 +67,8 @@ public class Block {
    }
 
    public static int getStateId(IBlockState var0) {
+      // Test
+
       Block var1 = var0.getBlock();
       return getIdFromBlock(var1) + (var1.getMetaFromState(var0) << 12);
    }
-- 
2.7.1.windows.2

