From fa1af4370fa4c3bb0c51387c2616006744a7035d Mon Sep 17 00:00:00 2001
From: Johannes Donath <johannesd@torchmind.com>
Date: Wed, 1 Jun 2016 21:08:19 +0200
Subject: [PATCH] Test commit ...


diff --git a/net/minecraft/block/Block.java b/net/minecraft/block/Block.java
index 47e28e6..8472b49 100644
--- a/net/minecraft/block/Block.java
+++ b/net/minecraft/block/Block.java
@@ -218,6 +218,8 @@ public class Block {
   }
 
   public static int getStateId(IBlockState var0) {
+    // Test
+
     Block var1 = var0.getBlock();
     return getIdFromBlock(var1) + (var1.getMetaFromState(var0) << 12);
   }
-- 
2.7.1.windows.2

